pyannote/audio/generators/change.py

```diff
@@ -0,0 +1,130 @@
+from pyannote.audio.generators.periodic import PeriodicFeaturesMixin
+from pyannote.core import SlidingWindowFeature
+from pyannote.core import Segment
+from pyannote.core import Timeline
+from pyannote.generators.fragment import SlidingSegments
+from pyannote.generators.batch import FileBasedBatchGenerator
+from scipy.stats import zscore
+import numpy as np
+
+
+class ChangeDetectionBatchGenerator(PeriodicFeaturesMixin,
+                                            FileBasedBatchGenerator):
+
+    """(X_batch, y_batch) batch generator
+
+    Yields batches made of subsequences obtained using a sliding
+    window over the audio files.
+
+    Parameters
+    ----------
+    feature_extractor : YaafeFeatureExtractor
+    duration: float, optional
+        yield semgents of length `duration`
+        Default to 3.2s.
+    step: float, optional
+        Duration and step of sliding window (in seconds).
+        Default to 0.8s.
+    balance: float, optional
+        Frames within a context window centred by change 
```

Comment:
Please replace this by something like:
`Artificially increase the number of positive labels by labeling as positive every frame in the direct neighborhood (less than `balance` seconds apart) of each change point. Defaults to 0.01s (10 ms).`

(I simply cut-and-pasted from the paper...)